src/ucs/stats/stats.c

```diff
@@ -112,6 +117,51 @@ static void ucs_stats_clean_node(ucs_stats_node_t *node) {
     ucs_list_del(&node->type_list);
 }
 
+static ucs_stats_class_t *ucs_stats_dup_class(ucs_stats_class_t *cls)
+{
+    ucs_stats_class_t *dup;
+
+    dup = ucs_malloc(sizeof(*cls) + sizeof(*cls->counter_names) * cls->num_counters,
+                     "ucs_stats_class_t");
+    if (dup == NULL) {
+        ucs_error("failed to allocate statistics class");
+    }
+
+    dup->name = ucs_strdup(cls->name, "ucs_stats_class_t name");
+    if (dup->name == NULL) {
+        ucs_error("failed to copy statistics class name");
```

Comment:
return NULL and free dup?
The same is relevant for another error below

Reply:
there is no way for error handling in statistics, we may just abort application on failures

Reply:
it's always better have clear error, than segfault. Also I'd clean the resources before returning error, because you can't be sure that the program is being closed (so memleak is possible). If you want to fail here, plz use `ucs_fatal`

Reply:
right, I missed that ucs_error doesn't abort application